agent_issuance/src/offer/aggregate.rs

```diff
@@ -91,6 +116,14 @@ impl Aggregate for Offer {
 
                 // If TxCode constraints are provided, generate a transaction code.
                 let tx_code = tx_code_constraints.as_ref().map(generate_tx_code);
+                if tx_code.is_some()
+                    && delivery_options
+                        .as_ref()
+                        .and_then(|opts| opts.recipient_email.as_ref())
+                        .is_none()
+                {
+                    return Err(MissingRecipientEmailError);
+                }
```

Comment:
Do we really want to enforce that the tx code can only be sent via email? Why would we not want to allow it that clients would create an offer that includes a TX code first, then they query (read) the Offer aggregate through the api and then deliver the tx code via their own means?